### PR TITLE
massive doc reorg/rewrite

### PR DESCRIPTION
**Pull request by SethTisue**
review by @adriaanm please. there are surely many details that can be improved (and some will be). what I'd like from the current round of review is just a sanity check that when I split it up into multiple files, everything ended up in the right place

### CODE REVIEW

**Comment by SethTisue** [comment]
review by @adriaanm please. there are surely many details that can be improved (and some will be). what I'd like from the current round of review is just a sanity check that when I split it up into multiple files, everything ended up in the right place

File: doc/overview.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+The new Jenkins infrastructure runs on virtual servers hosted by
+Amazon.  The virtual servers are created and configured automatically
+via Chef.  Everything is scripted and the scripts version-controlled
+so the servers can automatically be rebuilt at anytime.  This is all
+described in more detail below.
+
+### Pull request validation
+
+Every commit in a pull request (not just the last!) must pass a series
```
Comment (adriaanm) [inline doc/overview.md:73]: `[ci: last-only]` in the PR title tells scabot to only validate the last commit

File: doc/client-setup.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  User jenkins
+
+Host scabot
+  HostName jenkins-master
+  IdentityFile $PWD/.chef/scabot.pem
+  User scabot
+```
+
+Verify that you can actually ssh to the various machines.
+(But note that some of them aren't always up.)
```
Comment (adriaanm) [inline doc/client-setup.md:125]: only master is always up -- you can bring all nodes up by launching the associated worker on jenkins, which uses the https://github.com/typesafehub/ec2-start-stop jenkins plugin

**Comment by adriaanm** [comment]
LGTM! Thank you! This makes it much more digestible :-)

**Comment by SethTisue** [comment]
I made your suggested additions in 1101463c2971904ee5316127aae492b87ecbbc01.

**Pull request merged**